contracts/Oracle.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity ^0.5.16;
+
+import "./PriceOracle.sol";
+
+interface Feed {
+    function decimals() external view returns (uint8);
+    function latestAnswer() external view returns (uint);
+}
+
+contract Oracle is PriceOracle {
+
+    struct FeedData {
+        address addr;
+        uint8 tokenDecimals;
+    }
+
+    address public owner;
+    mapping (address => FeedData) public feeds; // cToken -> feed data
+    mapping (address => uint) public fixedPrices; // cToken -> price
+    uint8 constant DECIMALS = 36;
+
+    modifier onlyOwner {
+        require(msg.sender == owner, "ONLY OWNER");
+        _;
+    }
+
+    constructor() public {
+        owner = msg.sender;
+    }
+
+    function changeOwner(address owner_) public onlyOwner {
+        owner = owner_;
+    }
+
+    function setFeed(CToken cToken_, address feed_, uint8 tokenDecimals_) public onlyOwner {
+        feeds[address(cToken_)] = FeedData(feed_, tokenDecimals_);
```

Comment:
The feed is on the cToken? Not the underlying?

Reply:
it's the same. Just makes it easier to add ETH which is not a token